packages/date/src/convertFormatString.ts

```diff
@@ -211,15 +209,26 @@ const formatMap = {
   U: 't',
 };
 
+type FormatMap = typeof formatMap;
+type DateKey = keyof FormatMap;
+type ObtainKeys<Obj, Type> = {
+  [Prop in keyof Obj]: Obj[Prop] extends Type ? Prop : never;
+}[keyof Obj];
+type DateStringKey = ObtainKeys<FormatMap, string>;
+type DateFormatter =
+  | ((date: Date) => string | number)
```

Comment:
What does the pipe symbol in front of the line do? 🤓

Reply:
Apparently that's [how TypeScript multilines a union type](https://github.com/prettier/prettier/issues/3986) 🤷. Looks weird, gives me [SML vibes](https://en.wikipedia.org/wiki/Standard_ML#Declarative_definitions).